components/base/kinematicbase/execution.go

```diff
@@ -33,6 +32,7 @@ const (
 	// inputUpdateStepSeconds, and will correct if deviation is larger than this percent of that amount.
 	minDeviationToCorrectPct = 50.
 	microsecondsPerSecond    = 1e6
+	courseCorrectionMaxScore = 50. // Course correction solutions must score better than this
```

Comment:
[q]: does course correction obey bounding regions should they exist?

Reply:
It is aware of neither bounding regions nor obstacles. Obstacles are taken care of by transient obstacle avoidance, bounding regions should broadly be a non-issue as course correction will regress towards the planned path, not away from it.